pandas/tests/indexes/common.py

```diff
@@ -30,7 +31,7 @@
 class Base:
     """ base class for index sub-class tests """
 
-    _holder = None
+    _holder = None  # type: Optional[Type[IntervalIndex]]
```

Comment:
This should be Type[Index] and will fix other errors e.g.
```
pandas\tests\indexes\test_range.py:22: error: Incompatible types in assignment (expression has type "Type[RangeIndex]", base class "Base" defined the type as "None")
pandas\tests\indexes\timedeltas\test_timedelta.py:31: error: Incompatible types in assignment (expression has type "Type[TimedeltaIndex]", base class "Base" defined the type as "None")
pandas\tests\indexes\period\test_period.py:26: error: Incompatible types in assignment (expression has type "Type[PeriodIndex]", base class "Base" defined the type as "None")
pandas\tests\indexes\datetimes\test_datetimelike.py:10: error: Incompatible types in assignment (expression has type "Type[DatetimeIndex]", base class "Base" defined the type as "None")
```

Reply:
good call. thanks